### ISSUE TITLE
PouchDB promises not resolved in a WinJS app

### ISSUE THREAD

**Issue by nolanlawson** [pouchdb/pouchdb#3958]
I got a bug report via email:

Congrats for your awesome library, it’s very useful.
> 
> We’re are developing an Windows Store App with WinJS, and PouchDB. When the apps crashes, we catch all the errors in a global handler (window.onerror),
> 
> We handle the error and we don’t stop the app if the error isn’t critical. After the error, PouchDB still works, but no promise gets resolved. Do you know a solution?
> 
> Example:

```js
this.localDatabase.put(jsonObject).then(function (response) {

                            if (!response.ok) { //never gets called after an Javascript error on WinJS (the same with catch)
```

Here's the [test case](https://nolanlawson.s3.amazonaws.com/pouchdb/www/20150610/PouchdbPromises.zip), which is a WinJS app.

**Comment by calvinmetcalf** [comment]
how do you run it, it doesn't actually have the WinJS library, I'd suspect it's due to them catching the global errors which I'm guessing prevents lie from catching the error

**Comment by nolanlawson** [comment]
I don't know much about WinJS. I asked the reporter to comment in this thread. Another comment:

> Is a Windows 8 Store app with WinJS. I think is the same as Internet Explorer 11

**Comment by josete89** [comment]
Hello, I'm Ricardo teammate.

The error occurs when use a normal callback, localDB.put({},function(err,response){}) in other hand If use a promise localDB.put({}).then() then works fine.

Regards.

**Comment by calvinmetcalf** [comment]
so if there is an uncaught error somewhere else you use `window.onerror` to catch it, but after this happens using the callback interface `localDB.put({},function(err,response){}) ` no longer works but the promise interface does `localDB.put({}).then() ` ?

**Comment by calvinmetcalf** [comment]
if this is the case (especially if the error was thrown in a callback) then the issue can be fixed by compiling pouchdb with the most recent version of browserify (specifically 10.2.4 or higher) as this has the fix for process.nextTick breaking if you threw an error inside of process.nextTick

**Comment by nolanlawson** [comment]
fixed by 39e5367e71228884e0115ffde63664d2c910c415
